src/System.ServiceProcess.ServiceController/tests/TestServiceProvider.cs

```diff
@@ -53,6 +79,22 @@ public TestServiceProvider(string serviceName)
             CreateTestServices();
         }
 
+        public async Task<byte> ReadPipeAsync()
+        {
+            Task readTask;
+            byte[] received = new byte[] { 0 };
+            readTask = Client.ReadAsync(received, 0, 1);
+            await readTask.TimeoutAfter(readTimeout);
+            return received[0];
+        }
+
+        public byte GetBytes()
+        {
+            var task = Task.Run(() => ReadPipeAsync());
+            task.Wait();
+            return task.Result;
```

Comment:
>task.Result; [](start = 19, length = 12)

task.Result already waits for its completion, you could simply write: `GetBytes() => ReadPipeAsync().Result;`